### PR TITLE
Vendor latest distributions so we can use the new errcode stuff

### PR DESCRIPTION
**Pull request by duglin**
ping @stevvooe @RichardScothern

Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

**Comment by duglin** [comment]
ping @LK4D4

**Comment by tiborvass** [comment]
@duglin are there any changes to the docker code itself? or is that coming in another PR?

**Comment by duglin** [comment]
@tiborvass yes - if you look at the diffs you'll see a few changes in the "graph" dir due to a changed interface - the first 2 files listed. And then a new hash for 'distribution' itself in hack/vendor.sh

Beyond that all of the edits should be under /vendor/.../distribution

**Comment by tiborvass** [comment]
@duglin i see, I guess I'm used to having the vendor.sh PR in a separate commit :) would you mind doing that?

File: graph/pull_v2.go [hunk 1]
```diff
@@ -58,7 +59,11 @@ func (p *v2Puller) pullV2Repository(tag string) (err error) {
 		taggedName = utils.ImageReference(p.repoInfo.LocalName, tag)
 	} else {
 		var err error
-		tags, err = p.repo.Manifests().Tags()
+		manSvc, err := p.repo.Manifests(context.Background())
```
Comment (tiborvass) [inline graph/pull_v2.go]: @stevvooe @dmcgowan so FYI in many other places, we use a `nil` context, not sure if that's a problem.

**Comment by duglin** [comment]
@tiborvass done - I split them into two commits

File: graph/pull_v2.go [hunk 2]
```diff
@@ -58,7 +59,13 @@ func (p *v2Puller) pullV2Repository(tag string) (err error) {
 		taggedName = utils.ImageReference(p.repoInfo.LocalName, tag)
 	} else {
 		var err error
-		tags, err = p.repo.Manifests().Tags()
+
+		manSvc, err := p.repo.Manifests(context.Background())
```
Comment (tiborvass) [inline graph/pull_v2.go]: @stevvooe @dmcgowan so FYI in many other places, we use a nil context, not sure if that's a problem.

Comment (stevvooe) [inline graph/pull_v2.go]: @tiborvass That is probably incorrect.

Comment (tiborvass) [inline graph/pull_v2.go]: @stevvooe so you'd want to replace all nil contexts by `context.Background()` ?

Comment (stevvooe) [inline graph/pull_v2.go]: There should be _some_ context. `nil` is okay, for now.

Comment (tiborvass) [inline graph/pull_v2.go]: @stevvooe Then should we keep it all `nil` or it's fine to have it heterogeneous?

**Comment by duglin** [comment]
please don't merge this yet - there's one more change in distribution I'd like to add first.

**Comment by stevvooe** [comment]
@duglin I believe it is merged now.

File: graph/pull_v2.go (hunk 2 above)
Comment (stevvooe) [inline graph/pull_v2.go]: @tiborvass Diversity is always better.

**Comment by duglin** [comment]
@stevvooe @tiborvass latest 'distribution' code is now pulled in.

File: vendor/src/github.com/docker/distribution/registry/client/errors.go [hunk 3]
```diff
@@ -52,10 +52,14 @@ func handleErrorResponse(resp *http.Response) error {
 	if resp.StatusCode == 401 {
 		err := parseHTTPErrorResponse(resp.Body)
 		if uErr, ok := err.(*UnexpectedHTTPResponseError); ok {
-			return &errcode.Error{
-				Code:   v2.ErrorCodeUnauthorized,
-				Detail: uErr.Response,
-			}
+			return v2.ErrorCodeUnauthorized.WithDetail(uErr.Response)
+			/*
+				return &errcode.Error{
```
Comment (tiborvass) [inline vendor/src/github.com/docker/distribution/registry/client/errors.go]: dead code

Comment (duglin) [inline vendor/src/github.com/docker/distribution/registry/client/errors.go]: hmm wonder how we missed that - will fix.

**Comment by duglin** [comment]
grabbing latest "distribution" code that has the "dead code" removed.

**Comment by tiborvass** [comment]
LGTM

**Comment by duglin** [comment]
ping @stevvooe

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**